Undocumented breaking change in OpenID Connect configuration

The Ditto Release 1.0.0 introduced a breaking change in the OpenID Connect Configuration by adding an `https://` in-front of the configured URLs.

Configuration option: `-Dditto.gateway.authentication.oauth.openid-connect-issuers.myprovider`

In Ditto 1.0.0-M1a it still worked to have a URL with `https://auth.some.domain`. After the upgrade the following stack trace is thrown:

```
2019-12-18 09:56:11,124 WARN  [] o.e.d.s.g.s.a.TimeMeasuringAuthenticationProvider  - Failed to unwrap DittoRuntimeException from Throwable!
gateway_1        | java.util.concurrent.CompletionException: akka.http.scaladsl.model.IllegalUriException: Illegal URI reference: Invalid input '/', expected userinfo-char, pct-encoded, '@' or port (line 1, column 15): https://https://auth.some.domain/auth/realms/arealm/.well-known/openid-configuration
gateway_1        |               ^
gateway_1        | 	at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
gateway_1        | 	at java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
gateway_1        | 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
gateway_1        | 	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
gateway_1        | 	at java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
gateway_1        | 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(Unknown Source)
gateway_1        | 	at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
gateway_1        | 	at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
gateway_1        | Caused by: akka.http.scaladsl.model.IllegalUriException: Illegal URI reference: Invalid input '/', expected userinfo-char, pct-encoded, '@' or port (line 1, column 15): https://https://auth.some.domain/auth/realms/arealm/.well-known/openid-configuration
```

Fix
---

Either this has to be documented in the migration notes, or if this is considered a bug, it should be fixed by also accepting URLs containing https://. Depending on the solution the docs have to be searched and updated accordingly, since in some places the URLs contain `https://` prefix, which does not work at this moment.

--- a/documentation/src/main/resources/pages/ditto/release_notes_100.md
+++ b/documentation/src/main/resources/pages/ditto/release_notes_100.md
@@ -68,6 +68,10 @@ When closing a WebSocket session, a `NullPointerException` occurred which is fix
 
 ### Migration notes
 
+OpenID Connect URLs are now prefixes with `https://` per default. Any configured URLs containing `https://` will break the configuration.
+Instead of `https://auth.eclipse.de/auth/realms/ditto` it has to be `auth.eclipse.de/auth/realms/ditto` instead.
+The Configuration option is `ditto.gateway.authentication.oauth.openid-connect-issuers.myprovider`.
+
 Because we removed support for suffixed collections with this release, an offline migration with the provided script 
 is needed.
 

--- a/documentation/src/main/resources/pages/ditto/release_notes_100.md
+++ b/documentation/src/main/resources/pages/ditto/release_notes_100.md
@@ -68,7 +68,7 @@ When closing a WebSocket session, a `NullPointerException` occurred which is fix
 
 ### Migration notes
 
-OpenID Connect URLs are now prefixes with `https://` per default. Any configured URLs containing `https://` will break the configuration.
+OpenID Connect URLs are now prefixed with `https://` per default. Any configured URLs containing `https://` will break the configuration.
 Instead of `https://auth.eclipse.de/auth/realms/ditto` it has to be `auth.eclipse.de/auth/realms/ditto` instead.
 The Configuration option is `ditto.gateway.authentication.oauth.openid-connect-issuers.myprovider`.
 

